### PR TITLE
Delete JSONDiff and XMLDiff to test downstream apps

### PR DESCRIPTION
**Pull request by socratesgorilla**

### CODE REVIEW

**Comment by cticenhour** [comment]
Double-check your commit history, looks like my news commit (https://github.com/idaholab/moose/pull/21766/commits/a88e90c343c6f2c2e8a0f70d8bdd036a1a6e6f0d) merged in earlier today snuck in to this PR.

File: python/TestHarness/testers/SchemaDiff.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+from RunApp import RunApp
+from TestHarness import util
+import json, os
+
+class SchemaDiff(RunApp):
+    @staticmethod
+    def validParams():
+        params = RunApp.validParams()
+        params.addRequiredParam('schemadiff',   [], "A list of XML or JSON files to compare.")
+        params.addParam('gold_dir',      'gold', "The directory where the \"golden standard\" files reside relative to the TEST_DIR: (default: ./gold/). This only needs to be set if the gold file is in a non-standard location.")
```
Comment (dschwen) [inline python/TestHarness/testers/SchemaDiff.py:19]: It's "gold standard". See https://pubmed.ncbi.nlm.nih.gov/16402524/

Comment (socratesgorilla) [reply python/TestHarness/testers/SchemaDiff.py:19]: This is actually copied straight from the old XMLDiffer, I didn't write it. I can change it though.

Comment (dschwen) [reply python/TestHarness/testers/SchemaDiff.py:19]: yes, please, otherwise the git blame will still indicate that you are responsible for this :-)

**Pull request closed without merging**